magefile.go

```diff
@@ -1295,3 +1320,170 @@ func shouldBuildAgent() bool {
 	}
 	return ret
 }
+
+// Pre-requisite: user must have the gcloud CLI installed
+func authGCP(ctx context.Context) error {
+	// Check if gcloud CLI is installed
+	const cliName = "gcloud"
+	cmd := exec.CommandContext(ctx, "which", cliName)
```

Comment:
How would this work on windows? Are we not going to support devs that want to run integration tests on Windows? They should perform auth manually ? Should we use golang libraries to authenticate against gcloud/aws/azure APIs ?

Reply:
I think `gcloud` (the CLI) is [installable on Windows](https://cloud.google.com/sdk/docs/install#windows) too but, yes, we'd need to replace this `which` usage here — good catch.  

We could also consider using golang APIs but I'm not sure what downstream implications that has for the test runner.  I think we are currently working under the assumption that it will depend on the config/profile files created by these CLIs for auth tokens, selected projects, etc.

Let me start by trying to implementing the equivalent of `which` here for Windows.  If that doesn't do it, then we can consider the more involved option of replacing the CLIs with golang libraries.

Reply:
This should work on Windows, absolutely. I think we can push that to a follow up PR.

Reply:
I think the issue is with the `which` command, which doesn't exist on Windows.  But there is a `where` command in Windows that does the same thing, apparently, so I'll just use that here for Windows.

Reply:
Implemented the `which`/`where` change.  

As for the CLI vs. golang APIs, ++ to deferring this decision to a follow up PR.